src/components/molecules/Jukebox/Jukebox.tsx

```diff
@@ -0,0 +1,171 @@
+import { useProfileModalControls } from "hooks/useProfileModalControls";
+import { useUser } from "hooks/useUser";
+
+import { ButtonNG } from "components/atoms/ButtonNG";
+import { InputField } from "components/atoms/InputField";
+
+import "./Jukebox.scss";
+
+type JukeboxTypeProps = {
+  recentVenueUsers: readonly WithId<User>[];
+  updateIframeUrl: Dispatch<SetStateAction<string>>;
+  venue: WithId<AnyVenue>;
+};
+
+export const Jukebox: React.FC<JukeboxTypeProps> = ({
+  recentVenueUsers,
+  updateIframeUrl,
+  venue,
+}) => {
+  const { register, handleSubmit, watch, reset } = useForm<{
+    jukeboxMessage: string;
+  }>({
+    mode: "onSubmit",
+  });
+  const [isSendingMessage, setMessageSending] = useState(false);
+  const chatValue = watch("jukeboxMessage");
+  const { userId } = useUser();
+  const [filteredUser] = recentVenueUsers.filter(({ id }) => id === userId);
+  const tableRef = filteredUser?.data?.[venue.name]?.table;
+
+  const { sendJukeboxMsg, messagesToDisplay } = useJukeboxChat({
+    venueId: venue.id,
+    tableId: tableRef,
+  });
+  const filteredMessages = messagesToDisplay.filter(
+    ({ tableId }) => tableId === tableRef
+  );
+
+  useEffect(() => {
+    const [lastMessage] = messagesToDisplay.slice(-1);
+    let urlToEmbed = lastMessage?.text;
+
+    if (
+      urlToEmbed?.includes(YOUTUBE_SHORT_URL_STRING) &&
+      isValidUrl(urlToEmbed)
+    ) {
+      urlToEmbed = getYoutubeEmbedFromUrl(lastMessage?.text);
+    }
+
+    if (isValidUrl(urlToEmbed)) {
+      updateIframeUrl(urlToEmbed);
+    }
+  }, [messagesToDisplay, updateIframeUrl]);
+
+  const sendMessageToChat = handleSubmit(async ({ jukeboxMessage }) => {
+    setMessageSending(true);
+
+    await sendJukeboxMsg({ message: jukeboxMessage });
+    reset();
+  });
+
+  // @debt replace with useDebounce
+  // This logic disallows users to spam into the chat. There should be a delay, between each message
+  useEffect(() => {
+    if (!isSendingMessage) return;
+
+    const timeoutId = setTimeout(() => {
+      setMessageSending(false);
+    }, CHAT_MESSAGE_TIMEOUT);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [isSendingMessage]);
+
+  const isBtnDisabled = !chatValue || isSendingMessage;
+  const submitButtonClasses = classNames("Jukebox__buttons-submit", {
+    "Jukebox__buttons-submit--active": !isBtnDisabled,
+  });
+
+  const messagesEndRef = useRef<null | HTMLDivElement>(null);
+
+  const scrollToBottom = useCallback(() => {
+    messagesEndRef.current?.scrollIntoView({ behavior: "smooth" });
+  }, []);
+
+  useEffect(scrollToBottom, [messagesToDisplay, scrollToBottom]);
+
+  const { openUserProfileModal } = useProfileModalControls();
+
+  const jukeboxChatMessages = useMemo(
+    () =>
+      filteredMessages?.map((msg) => {
+        const isUrl = isValidUrl(msg.text);
+        return (
+          <div key={msg.id} className="Jukebox__chat-messages">
+            <span
+              className="Jukebox__chat-author"
```

Comment:
let's make them more obvious to click. This class name makes them look like URLs adding yellow color on hover.
```suggestion
              className="Jukebox__chat-author button--a"
```